extensions/cms/src/cmsUtils.ts

```diff
@@ -24,9 +31,35 @@ const mssqlProvider: string = 'MSSQL';
  */
 export class CmsUtils {
 
+	private _credentialProvider: azdata.CredentialProvider;
 	private _cmsService: mssql.CmsService;
 	private _registeredCmsServers: ICmsResourceNodeInfo[];
 
+	constructor() {
+		// 1) Get a credential provider
+		// 2) Store the credential provider for use later
+		this.setupCredentialProvider();
+	}
+
+	private async setupCredentialProvider(): Promise<azdata.CredentialProvider> {
+		if (!this._credentialProvider) {
+			this._credentialProvider = await azdata.credentials.getProvider(CredentialNamespace);
+		}
+		return this._credentialProvider;
+	}
+
+	public async savePassword(username: string, password: string): Promise<boolean> {
+		let provider = await this.credentialProvider();
```

Comment:
Please try to get in the habit of using const instead of let wherever you can - it helps declare intent (is this var intended to be changed or not?)